proj-android/PowerAuthLibrary/src/main/java/io/getlime/security/powerauth/keychain/fingerprint/FingerprintAuthenticationHandler.java

```diff
@@ -246,21 +236,31 @@ public FingerprintStage initCrypto() {
         }
     }
 
+    /**
+     * Contains cached encrypted key. The value is calculated only for once.
+     */
+    private byte[] alreadyEncryptedKey;
+
+    /**
+     * Contains true if {@link #alreadyEncryptedKey} has been already constructed.
+     * The purpose of this flag is to prevent duplicate calls to {@code mCipher.doFinal()}.
+     */
+    private boolean hasAlreadyEncryptedKey;
```

Comment:
Do we need this property, or can we just work with `alreadyEncryptedKey`? As a syntactic sugar, we can have:

```java
private boolean hasAlreadyEncryptedKey() {
    return alreadyEncryptedKey != null;
}
```

Reply:
I wanted to avoid double call to `doFinal()` on `Cipher` (original source of crash). It can be reproduced by double "success" reported by the `FingerprintManager`. I guess that some devices can report multiple "success" states quickly in relative short time. 

This boolean guarantees that it's not called for twice even if the first attempt failed and the result is `null`. Perhaps it's overcomplicated. 

This is completely reworked In upcoming `1.2.0`, so the object receiving events from `FingerprintManager` will never report this situation for twice.

Reply:
OK, understood...